### PR TITLE
18 model properties

### PR DESCRIPTION
**Pull request by liunelson**
Added (or moved) "name" and "description" properties, wherever missing, to the model object property list of the Petri Net and Reg Net frameworks. Also updated the examples.

### CODE REVIEW

File: regnet/examples/syntax_edge_cases.json
Comment (mehalter) [inline regnet/examples/syntax_edge_cases.json:32]: This is incorrect, `name` and `description` should be in the `properties` of the edge because it provides no structural meaning to the edge. The `vertices` do not have a nested `properties` field because there are no real structure of a and so all keys under it are actually just a `properties` table basically.

This does expose issues in our JSON validation because this should not validate. I will look into this soon to resolve this bug in our JSON Schema

File: regnet/regnet_schema.json
Comment (mehalter) [inline regnet/regnet_schema.json:80]: Oh I see, you changed the validation. I think that this is incorrect and should not be done this way. It begins to intermingle the idea of "metadata" and "model structure" which we have been separating with this model representation.

**Review by mehalter — CHANGES_REQUESTED** [review]
I left a few comments. The suggested changes break our separation of metadata and model structure which would be a very very significant change in methodology and semantics of this model representation and needs further discussion if we want to move forward with this.

Also the `syntax_edge_cases.json` file should be left as it is. It is intentionally missing data because all of these fields are strictly not required to have a valid model. It is there to make sure that models with these missing fields or partially present fields are still 1. validated by our JSON Schema and 2. can be used for performers to test that their serialization/deserialization scripts actually work against the JSON Schema rather than people assuming these optional fields to exist.

File: regnet/examples/syntax_edge_cases.json
Comment (mehalter) [reply regnet/examples/syntax_edge_cases.json:32]: I realized later in the review that our original JSON Schema was correct and changes were made to it to make this validate. The 2nd half of this comment can be ignored as it was brought back up in a separate conversation here.

Comment (bgyori) [reply regnet/examples/syntax_edge_cases.json:32]: I think we could follow what @mehalter is saying here i.e., name and description could remain inside properties. I don't have a strong opinion either way though.

**Comment by liunelson** [comment]
@mehalter Thanks for the review. I'll hold off on this issue though until the refactor PR gets merged.

File: regnet/examples/syntax_edge_cases.json
Comment (liunelson) [reply regnet/examples/syntax_edge_cases.json]: I think I understand what you mean. I've moved `name` and `description` into `properties` of both RegNet edges and PetriNet transitions.

File: regnet/regnet_schema.json
Comment (liunelson) [reply regnet/regnet_schema.json:80]: Your original approach makes sense: I've changed both `petrinet_schema` and `regnet_schema` to keep `name` and `description` of RegNet edges and PetriNet transitions in `properties`.

**Comment by liunelson** [comment]
@mehalter I've also reverted `syntax_edge_cases.json` back to the original.

> Also the `syntax_edge_cases.json` file should be left as it is. It is intentionally missing data because all of these fields are strictly not required to have a valid model. It is there to make sure that models with these missing fields or partially present fields are still 1. validated by our JSON Schema and 2. can be used for performers to test that their serialization/deserialization scripts actually work against the JSON Schema rather than people assuming these optional fields to exist.

**Comment by liunelson** [comment]
@mehalter I've made all the changes that you pointed out. If no other issue, this PR should be good to be merged.

**Comment by mehalter** [comment]
I'll review it again tomorrow.

**Review by mehalter — CHANGES_REQUESTED** [review]
Looks great! I agree with the changes that were made. Could you do the rebase on the new `main` branch after the refactor and I can take another look at it just as a sanity check and get this merged in. Thanks @liunelson !

**Comment by liunelson** [comment]
> Looks great! I agree with the changes that were made. Could you do the rebase on the new `main` branch after the refactor and I can take another look at it just as a sanity check and get this merged in. Thanks @liunelson !

This PR can be closed since I accidentally included these changes in the "type info" PR.

**Pull request closed without merging**
